src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(5);
+        switch (figureType) {
+            case 0:
+                return new Square(random.nextDouble() * 10,
+                        colorSupplier.getRendomColor());
```

Comment:
All magic numbers, strings, symbols in your code should be constants.
https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names
You can move this logic to private method
```suggestion
                return prepareSquere()
```